improver/cli/__init__.py

```diff
@@ -387,6 +405,106 @@ def improver_help(prog_name: parameters.pass_name, command=None, *, usage=False)
     return result
 
 
+# run-workflow command
+
+
+def command_executor(output, *argv, verbose=False, dry_run=False):
+    if argv[0] == "improver":
+        result = execute_command(
+            SUBCOMMANDS_DISPATCHER, *argv, verbose=verbose, dry_run=dry_run
+        )
+        if result is None:
+            # assume it's sufficient to pass along the output instead
+            result = output
+            if verbose or dry_run:
+                print(output)
+    else:  # assume generic shell command
+        cmd = " ".join(argv)
+        if verbose or dry_run:
+            print(cmd)
+        if not dry_run and os.system(cmd):  # nosec
```

Comment:
There are concerns over `os.system` calls here from BOM.
The issue seems to be that it is executed within improver, rather than as part of the suite code.

- One option could be to call python functions instead (whether replacing system calls with python equivalents where possible or otherwise wrapping subprocess calls where required).
- And/or perhaps we can consider how this might be migrated to paraflow in some way.  I have concerns over this for separation of concerns.

Something to consider.

Reply:
Yeah, I suspected this might get flagged up. I'll leave that question up to the interested parties to sort out 😁.

Other solutions then you mentioned:
- Add config system (could be useful in its own right), where certain features and be enabled/disabled (this won't silence the mindless security bots, though).
- Add extension entry points for 3-rd party packages to use and move that functionality out of the codebase (might be a pain to manage though). Could be done also in a bit less principled way by allowing to drop some extra python module(s) in the right place during build/deployment.